### PR TITLE
update dependency installer for Arch

### PR DESCRIPTION
**Pull request by leongross**
Update `update_dependecy.sh` for `Arch Linux` to use igraph version `0.9.9-1`

### CODE REVIEW

File: install_dependencies.sh [hunk 1]
```diff
@@ -79,33 +79,80 @@ elif [[ "$platform" == 'linux' ]]; then
...
-        qt5-base python ccache autoconf libsodium igraph qt5-svg ninja lcov \
-        gcovr python-sphinx doxygen python-sphinx_rtd_theme python-jedi \
-        python-pip pybind11 rapidjson spdlog graphviz boost \
-        python-dateutil z3
+            qt5-base python ccache autoconf libsodium qt5-svg ninja lcov \
+            gcovr python-sphinx doxygen python-sphinx_rtd_theme python-jedi \
+            python-pip pybind11 rapidjson spdlog graphviz boost \
+            python-dateutil z3
+
+        # until igraph version 10 is supported, backport the igraph version on Arch Linux
```
Comment (szhorvat) [inline install_dependencies.sh:98]:
```suggestion
        # until igraph version 0.10 is supported, backport the igraph version on Arch Linux
```

**Comment by szhorvat** [comment]
You may want to warn users that downgrading igraph will break other packages, notably python-igraph.

**Comment by joern274** [comment]
Following the add_subdirectory suggestion I would like to offer an alternative solution how to provide igraph-0.9.10 for all supported platforms. Please have a look at the HAL branch **fix/igraph_subdirectory**

@leongross , can you please verify that this branch builds and runs on your Arch system ?

@szhorvat , do you have time to review the changes? There is at least one issue I am not so happy about : when downloading the igraph-0.9.10.tar.gz tarball from github it will only build after patching the version number at two different locations. Did I miss something or is that intended by maintainer?

**Comment by szhorvat** [comment]
https://github.com/igraph/igraph/archive/refs/tags/0.9.10.tar.gz is just a snapshot of the repository generated by GitHub automatically. When building igraph from the contents of the repository, it is necessary for an actual git repoto be backing it, otherwise the version number detection will fail.

If you are aiming to use specifically version 0.9.10, then download the _release tarball_:

https://github.com/igraph/igraph/releases/download/0.9.10/igraph-0.9.10.tar.gz

Note the difference: you need `igraph-0.9.10.tar.gz`, not `0.9.10.tar.gz`.

An alternative is to use a git submodule. Then you can even use the most recent `release/0.9` branch, which has some extra fixes on top of 0.9.10. These fixes were necessary for the R and Mathematica interfaces of igraph, but they are not released within the 0.9 series (of course the 0.10 series contains them).

**Comment by szhorvat** [comment]
Another comment is that instead of _installing_ igraph in a temporary location, it should be possible to use [CMake's `add_subdirectory`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html).

There is an example that shows this here: https://github.com/igraph/igraph-example/tree/master/cmake-project-2

I think this would be a simpler solution for you. Hopefully, there should be no issues.

I should say though that I am a physicist, not a programmer, so my knowledge of arcane technical details only goes so far 🙂

**Comment by szhorvat** [comment]
> There is still the need to patch `src/CMakeLists.txt` unless we are fine with building `libigraph.so.0.0.0` without correct version number. Of course not a big deal.

The SONAME is set properly only starting with igraph 0.10. If you need to set it for 0.9, please use the name `libigraph.so.2.0.0` to stay consistent with Debian/Ubuntu. For 0.10, we use `libigraph.so.3.0.0`.

**Pull request closed without merging**